### DISCUSSION TITLE
Purchasing boards

### DISCUSSION THREAD

**Discussion by bb12489** [nealcarney/ESPHome-MaxxFan-controller#3] [General]
I'm unfamiliar with the site that is hosting your designs. Can they produce these boards complete with all components already soldered on? Also would this work for a Maxxair fan that doesn't have IR or the lift motor? I just want to be able to control the on/off and fan speeds. My fans do have the RJ45 ports on them though.

**Comment by nealcarney** [comment]
The design files are located at https://oshwlab.com/ncarney/maxxfan-controller, which is on OSHWLab, an open-source PCB design library.  That site is owned/operated by EasyEDA (https://easyeda.com/), which is a Chinese web-based PCB design tool.  You can set up an account in EasyEDA and import designs from OSHWLab into your account. Once you finish your design, you can indeed have their affiliated company produce either a raw PCB or a finished board the components populated.  That company is JLCPCB (https://jlcpcb.com/), but it is fully integrated with the EasyEDA workflow.  I have only used the raw PCB service and they do a terrific job.

You could theoretically get them to produce my design with components.  However, I'm not sure they stock the various components.  Also, my board design shows several modules directly soldered onto the PCB (like the ESP32, the INA226 and the Mini360) whereas in reality I soldered female headers onto the PCB and then just inserted those modules into the header in case I needed to replace them.

Concerning whether this would work on your particular Maxxair fan, I'm not sure it would.  Not having the IR would make the IR stuff on my board unnecessary but would not necessarily prevent it from working.  However the pinout for the MaxxFan RJ45 connections may not be the same, and my guess is that they probably aren't since you don't have the lift motor (for sure that would be missing).  Below is a link to the pinouts for the "K" model fans.  You would want to check and make sure that the pinouts for your particular model are the same, or obtain the relevant pinouts.  Can do either via Google search of contact MaxxFan/Airxcel directly.
[Pin Out for K-Model MaxxFans_May2013.pdf](https://github.com/nealcarney/ESPHome-MaxxFan-controller/files/14004681/Pin.Out.for.K-Model.MaxxFans_May2013.pdf)

**Reply by bb12489** [reply]
Thanks for the insight. I'll look more into their site to see about getting some boards produced. Having the female adapters is a good idea to be able to replace things. As for the fans, I think I may be in luck. My models are the 5100K, so I'm thinking the pin out is the same. I often wondered if I could add the lift motor to mine without much issues. Maybe even the IR functions as well.

**Reply by bb12489** [reply]
@nealcarney Do you have a list of where you purchased all the components?

**Reply by nealcarney** [reply]
I don't, but I'll try to put one together in the next few days.  Traveling right now.

**Reply by bb12489** [reply]
Thanks so much. I ordered some pcbs, and am going to try my hand at adding the lift motor and gear box to my fan. I really only need to be able to turn the fan on, change speeds, and open/close the lid.

**Reply by nealcarney** [reply]
I posted an excel file (MaxxFan-and-Houghton-AC-Controller_components.xlsx) with the components.  It is located in the misc info folder.  I hope this helps.

**Reply by bb12489** [reply]
This is great. Thank you! I just got the pcbs in the mail, so I'll see if I can get this to work. Thank you for the help.
